packages/test-utils/event-assertion/src/lib.rs

```diff
@@ -24,11 +39,35 @@ impl<'a> EventAssertion<'a> {
             _ => Symbol::new(self.env, symbol_name),
         };
 
-        events.iter().find(|e| {
-            let topics: Vec<Val> = e.1.clone();
+        for (event_index, event) in events.iter().enumerate() {
+            let topics: Vec<Val> = event.1.clone();
+            if topics.is_empty() {
+                continue;
+            }
+            
             let topic_symbol: Symbol = topics.first().unwrap().into_val(self.env);
-            topic_symbol == target_symbol
-        })
+            if topic_symbol == target_symbol {
+                let event_key = (event_index as u32, 0);
```

Comment:
what does `0` corresponds to here exactly? Why are we assigning `0` to every event?

Reply:
Adding it later wouldn't be a big problem, so let's keep it as simple as we can for now.